common/rpc/interceptor/namespace_validator.go

```diff
@@ -103,6 +111,59 @@ func (ni *NamespaceValidatorInterceptor) LengthValidationIntercept(
 	return handler(ctx, req)
 }
 
+func (ni *NamespaceValidatorInterceptor) setNamespaceIfNotPresent(
+	namespaceRegistry namespace.Registry,
+	req interface{},
+) error {
+	namespaceName := GetNamespace(namespaceRegistry, req)
+	if namespaceName != "" {
+		return nil
+	}
```

Comment:
Can we check if req is a NamespaceGetter and if request.GetNamespace() is empty? I feel it's hard to reason about `GetNamespace` function which hides the error.

Then we also don't need the empty check in `setNamespace`